collectorist/api/src/server.rs

```diff
@@ -0,0 +1,36 @@
+use std::net::SocketAddr;
+
+use actix_web::{
+    middleware::{Compress, Logger},
+    web, App, HttpServer,
+};
+use derive_more::{Display, Error, From};
+use utoipa::OpenApi;
+use utoipa_swagger_ui::SwaggerUi;
+
+use crate::SharedState;
+
+#[derive(OpenApi)]
+#[openapi(paths())]
+pub struct ApiDoc;
+
+pub async fn run<B: Into<SocketAddr>>(state: SharedState, bind: B) -> Result<(), anyhow::Error> {
+    let openapi = ApiDoc::openapi();
+
+    let addr = bind.into();
+    log::debug!("listening on {}", addr);
+    HttpServer::new(move || {
+        App::new()
+            .wrap(Logger::default())
+            .wrap(Compress::default())
+            .app_data(web::Data::new(state.clone()))
+            .service(SwaggerUi::new("/swagger-ui/{_:.*}").url("/openapi.json", openapi.clone()))
+    })
+    .bind(addr)?
+    .run()
+    .await?;
+    Ok(())
```

Comment:
You're creating an HttpServer that seems to do nothing? This is where I'd expect your listener loop to be. It'd be listening for csub stuff instead of HTTP requests.